pkg/ring/lifecycler.go

```diff
@@ -564,11 +585,12 @@ func (i *Lifecycler) autoJoin(ctx context.Context, targetState IngesterState) er
 
 		newTokens := GenerateTokens(i.cfg.NumTokens-len(myTokens), takenTokens)
 		i.setState(targetState)
+
 		ringDesc.AddIngester(i.ID, i.Addr, newTokens, i.GetState(), i.cfg.NormaliseTokens)
 
-		tokens := append(myTokens, newTokens...)
-		sort.Sort(sortableUint32(tokens))
-		i.setTokens(tokens)
+		myTokens = append(myTokens, newTokens...)
```

Comment:
This code deals with similar situation than saving/loading tokens -- existing tokens in the ring, that belong to the current ingester. Perhaps this can be reused instead of using files?

(There is of course a bug in that AddIngester should not be called with `newTokens` only, but with `myTokens`)

Reply:
I don't think I understand. The current flow is to read the file only if we don't exist in the ring yet. So if do exist in the ring, we are re-using those tokens and we never read from the file.

Reply:
This was just another comment about the same suggestion of reusing ring (instead of writing to file), to which you have already replied. [If you want to fix call to `AddIngester` with wrong tokens, feel free, but if not, it doesn't like anyone is complaining about is]